csharp/src/Ice/Proxy.cs

```diff
@@ -252,6 +269,80 @@ public static class Proxy
+
+            static async Task<T> ReadResponseAsync(
+                ValueTask<IncomingResponseFrame> task,
+                InputStreamReader<T> reader,
+                Communicator communicator) =>
+                (await task.ConfigureAwait(false)).ReadReturnValue(communicator, reader);
+        }
+
+        /// <summary>Sends a request that returns void and waits synchronously for the result.</summary>
+        /// <param name="proxy">The proxy for the target Ice object.</param>
+        /// <param name="request">The outgoing request frame for this invocation. Usually this request frame should have
+        /// been created using the same proxy, however some differences are acceptable, for example proxy can have
+        /// different endpoints.</param>
+        /// <param name="oneway">When true, the request is sent as a oneway request. When false, it is sent as a
+        /// twoway request.</param>
+        /// <param name="cancel">A cancellation token that receives the cancellation requests.</param>
+        public static void InvokeVoid(
+            this IObjectPrx proxy,
+            OutgoingRequestFrame request,
+            bool oneway,
+            CancellationToken cancel = default)
+        {
+            IncomingResponseFrame response = proxy.Invoke(request, oneway, cancel);
+            if (!oneway)
+            {
+                response.ReadVoidReturnValue(proxy.Communicator);
+            }
+        }
+
+        /// <summary>Sends a request that returns void and returns the result asynchronously.</summary>
+        /// <param name="proxy">The proxy for the target Ice object.</param>
+        /// <param name="request">The <see cref="OutgoingRequestFrame"/> for this invocation. Usually this request
+        /// frame should have been created using the same proxy, however some differences are acceptable, for example
+        /// proxy can have different endpoints.</param>
+        /// <param name="oneway">When true, the request is sent as a oneway request. When false, it is sent as a
+        /// twoway request.</param>
+        /// <param name="progress">Sent progress provider.</param>
+        /// <param name="cancel">A cancellation token that receives the cancellation requests.</param>
+        public static Task InvokeVoidAsync(
+            this IObjectPrx proxy,
+            OutgoingRequestFrame request,
+            bool oneway,
+            IProgress<bool>? progress = null,
+            CancellationToken cancel = default)
+        {
+            ValueTask<IncomingResponseFrame> response = proxy.InvokeAsync(request, oneway, progress, cancel);
+            return oneway ? Task.CompletedTask : ReadResponseAsync(response, proxy.Communicator);
+
+            static async Task ReadResponseAsync(
+                ValueTask<IncomingResponseFrame> response,
+                Communicator communicator) => (await response.ConfigureAwait(false)).ReadVoidReturnValue(communicator);
```

Comment:
Same here, it could be the following:
```
        public static async Task InvokeVoidAsync(
            this IObjectPrx proxy,
            OutgoingRequestFrame request,
            bool oneway,
            IProgress<bool>? progress = null,
            CancellationToken cancel = default)
        {
            ValueTask<IncomingResponseFrame> response = proxy.InvokeAsync(request, oneway, progress, cancel);
            if (!oneway)
            {
                (await response.ConfigureAwait(false)).ReadVoidReturnValue(proxy.Communicator);
            }
        }
```

Reply:
See above.